### PR TITLE
Enable IAM role-based auth between EC2 and S3

### PR DESCRIPTION
**Pull request by zekehuntergreen**
resolves https://github.com/alephdata/aleph/issues/3185

- ~bump version of servicelayer to include changes in https://github.com/alephdata/servicelayer/pull/99~
- add comment explaining that role-based auth is to be preferred over user-based, and that environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are only required for user-based auth.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ followthemoney==3.4.3
 followthemoney-store[postgresql]==3.0.5
 followthemoney-compare==0.4.4
 fingerprints==1.0.3
-servicelayer[google,amazon]==1.21.0
+https://github.com/guardian/aleph-servicelayer/archive/iam-role-based-s3-auth.zip
```
Comment (zekehuntergreen) [inline requirements.txt:7]: TODO: update when https://github.com/alephdata/servicelayer/pull/99 is released

File: aleph.env.tmpl [hunk 2]
```diff
@@ -63,9 +63,11 @@ ALEPH_OAUTH_SECRET=
 # Or, if 'ALEPH_ARCHIVE_TYPE' configuration is 's3':
 # ARCHIVE_TYPE=s3
 # ARCHIVE_BUCKET=
+# AWS_REGION=
+# If server is running on EC2, prefer role-based authentication and leave key ID and access key null.
```
Comment (stchris) [inline aleph.env.tmpl:67]: Here's a suggestion which includes the link to the boto docs:

```suggestion
# Leave these next two keys empty if you prefer IAM Role-based auth
# (see https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#id1)
```

File: aleph.env.tmpl [hunk 3]
```diff
@@ -63,9 +63,11 @@ ALEPH_OAUTH_SECRET=
...
+# Otherwise, provide key ID and access key.
```
Comment (stchris) [inline aleph.env.tmpl:68]: Perhaps we can leave this line out?

```suggestion

```

**Review by stchris — APPROVED** [review]
Perfect!

**Pull request merged**